src/sentry/integrations/slack/actions/notification.py

```diff
@@ -122,41 +128,56 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
+                    and rule_id
+                    and OrganizationOption.objects.get_value(
+                        organization=self.project.organization,
+                        key="sentry:issue_alerts_thread_flag",
+                        default=ISSUE_ALERTS_THREAD_DEFAULT,
+                    )
+                ):
+                    return None
+
                 try:
                     parent_notification_message = self._repository.get_parent_notification_message(
                         rule_id=rule_id,
                         group_id=event.group.id,
                         rule_action_uuid=rule_action_uuid,
                     )
-                except Exception:
+                except Exception as e:
+                    lifecycle.record_halt(e)
+
                     # if there's an error trying to grab a parent notification, don't let that error block this flow
                     # we already log at the repository layer, no need to log again here
-                    pass
+                    return None
 
-                if parent_notification_message:
-                    # If a parent notification exists for this rule and action, then we can reply in a thread
-                    # Make sure we track that this reply will be in relation to the parent row
-                    new_notification_message_object.parent_notification_message_id = (
-                        parent_notification_message.id
-                    )
-                    # To reply to a thread, use the specific key in the payload as referenced by the docs
-                    # https://api.slack.com/methods/chat.postMessage#arg_thread_ts
-                    thread_ts = parent_notification_message.message_identifier
-                    # If this flow is triggered again for the same issue, we want it to be seen in the main channel
-                    reply_broadcast = True
+                if parent_notification_message is None:
+                    return None
+
+                # If a parent notification exists for this rule and action, then we can reply in a thread
+                # Make sure we track that this reply will be in relation to the parent row
+                new_notification_message_object.parent_notification_message_id = (
+                    parent_notification_message.id
+                )
+                # To reply to a thread, use the specific key in the payload as referenced by the docs
+                # https://api.slack.com/methods/chat.postMessage#arg_thread_ts
+                return parent_notification_message.message_identifier
+
+            with MessagingInteractionEvent(
+                MessagingInteractionType.GET_PARENT_NOTIFICATION, SlackMessagingSpec()
+            ).capture() as lifecycle:
+                thread_ts = get_thread_ts(lifecycle)
+
+            # If this flow is triggered again for the same issue, we want it to be seen in the main channel
+            reply_broadcast = thread_ts is not None
```

Comment:
This simplification (settings `reply_broadcast` based on whether we got a non-null `thread_ts` rather than as part of the if-else flow) assumes that we would never expect to get a null value for `parent_notification_message.message_identifier` in line 157 of the old solution. This is possible according to the type annotations on `IssueAlertNotificationMessage` but I don't think it can happen in practice.

The comment seems to back this up (we want to broadcast if and only if we are replying to a thread) so AFAICT this would be more correct behavior anyway in the case where `parent_notification_message.message_identifier` is null.

Reply:
I agree with your change.

> but I don't think it can happen in practice

Is the case where parent is not valid anymore because of slack retention reasons one of that? Or this parent is only a sentry object.

Reply:
Good question. I don't think so? My assumption would be that `self._repository.get_parent_notification_message` would return a null value if the message had been deleted. My changes would affect the value of `reply_broadcast` only if a non-null `parent_notification_message` value were returned, but with a null `message_identifier` attribute.

> Or this parent is only a sentry object.

Could you elaborate? Would that be a case where `parent_notification_message.message_identifier` is None, to your knowledge?

Regardless, I'm fairly confident in saying that `reply_broadcast = thread_ts is not None` is the business logic we want anyway.